code/game/objects/effects/spawners/gibspawner.dm

```diff
@@ -21,82 +21,81 @@
 	var/fleshcolor //Used for gibbed humans.
 	var/bloodcolor //Used for gibbed humans.
 
-	New(location, var/fleshcolor, var/bloodcolor)
-		..()
+/obj/effect/spawner/gibspawner/Initialize(mapload, fleshcolor, bloodcolor)
+	..()
 
-		if(fleshcolor) src.fleshcolor = fleshcolor
-		if(bloodcolor) src.bloodcolor = bloodcolor
+	if(fleshcolor)
+		src.fleshcolor = fleshcolor
+	if(bloodcolor)
+		src.bloodcolor = bloodcolor
 
-		if(istype(loc,/turf)) //basically if a badmin spawns it
-			Gib(loc)
+	if(istype(loc,/turf)) //basically if a badmin spawns it
+		Gib(loc)
+	return INITIALIZE_HINT_QDEL
 
-	proc/Gib(atom/location)
-		if(gibtypes.len != gibamounts.len || gibamounts.len != gibdirections.len)
-			to_chat(world, span_warning("Gib list length mismatch!"))
-			return
+/obj/effect/spawner/gibspawner/proc/Gib(atom/location)
```

Comment:
This code is trash cannot we just remove it

Reply:
iirc this is used for mapping